src/py/pyodide/utils/abort/__init__.py

```diff
@@ -0,0 +1,35 @@
+from __future__ import annotations
+
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from collections.abc import Callable, Iterable
+
+    from js import AbortSignal
+
+
+_abort_signal_any: Callable[[Iterable[AbortSignal]], AbortSignal] | None = None
+
+
+def get_abort_signal_any() -> Callable[[Iterable[AbortSignal]], AbortSignal]:
+    from js import AbortSignal
+
+    global _abort_signal_any
+
+    if AbortSignal.any is None:
+        from ...code import run_js  # type:ignore[unreachable]
+
+        path = Path(__file__).parent / "polyfill.js"
+        source = path.read_text() + ";polyfillAbortSignal()"
+        _abort_signal_any = run_js(source)
```

Comment:
We don't want to use `run_js` here because some people need to run Pyodide in an environment where `eval` is disabled (by content security protocol or otherwise). See how we handled it in `scheduler.ts`:
https://github.com/pyodide/pyodide/blob/main/src/js/scheduler.ts
https://github.com/pyodide/pyodide/blob/main/src/js/api.ts?plain=1#L62
https://github.com/pyodide/pyodide/blob/main/src/py/pyodide/webloop.py?plain=1#L14

I think the reason that `scheduler.ts` exports and it's attached to `API` in `api.ts` is for typescript unit testing. If we don't need typescript unit tests we can attach it directly to API in `abortSignalAny.ts`.

Reply:
But we might as well keep the same pattern where we import in `api.ts` and attach it to API there.

Reply:
I attached it in `abortSignalAny.ts` and imported this script in `api.ts`. I can use the other way of course

Reply:
Thanks!